src/test/java/com/github/mtakaki/credentialstorage/hibernate/UnitOfWorkApplicationListenerTest.java

```diff
@@ -0,0 +1,397 @@
+
+    private final RequestEvent requestStartEvent = mock(RequestEvent.class);
+    private final RequestEvent requestMethodStartEvent = mock(RequestEvent.class);
+    private final RequestEvent responseFiltersStartEvent = mock(RequestEvent.class);
+    private final RequestEvent requestMethodExceptionEvent = mock(RequestEvent.class);
+    private final Session session = mock(Session.class);
+    private final Session analyticsSession = mock(Session.class);
+    private final Transaction transaction = mock(Transaction.class);
+    private final Transaction analyticsTransaction = mock(Transaction.class);
+
+    @Before
+    public void setUp() throws Exception {
+        this.listener.registerBundle(RemoteCredentialHibernateBundle.DEFAULT_NAME, this.bundle);
+        this.listener.registerBundle("analytics", this.analyticsBundle);
+
+        final SessionHolders sessionHolders = mock(SessionHolders.class);
+
+        when(this.bundle.getSessionHolders()).thenReturn(sessionHolders);
+        when(this.analyticsBundle.getSessionHolders()).thenReturn(sessionHolders);
+        when(this.bundle.getSessionFactory()).thenReturn(this.sessionFactory);
+        when(this.analyticsBundle.getSessionFactory())
+                .thenReturn(this.analyticsSessionFactory);
+
+        when(this.sessionFactory.openSession()).thenReturn(this.session);
+        when(this.session.getSessionFactory()).thenReturn(this.sessionFactory);
+        when(this.session.beginTransaction()).thenReturn(this.transaction);
+        when(this.session.getTransaction()).thenReturn(this.transaction);
+        when(this.transaction.isActive()).thenReturn(true);
+
+        when(this.analyticsSessionFactory.openSession()).thenReturn(this.analyticsSession);
+        when(this.analyticsSession.getSessionFactory()).thenReturn(this.analyticsSessionFactory);
+        when(this.analyticsSession.beginTransaction()).thenReturn(this.analyticsTransaction);
+        when(this.analyticsSession.getTransaction()).thenReturn(this.analyticsTransaction);
+        when(this.analyticsTransaction.isActive()).thenReturn(true);
+
+        when(this.appEvent.getType()).thenReturn(ApplicationEvent.Type.INITIALIZATION_APP_FINISHED);
+        when(this.requestMethodStartEvent.getType())
+                .thenReturn(RequestEvent.Type.RESOURCE_METHOD_START);
+        when(this.responseFiltersStartEvent.getType())
+                .thenReturn(RequestEvent.Type.RESP_FILTERS_START);
+        when(this.requestMethodExceptionEvent.getType()).thenReturn(RequestEvent.Type.ON_EXCEPTION);
+        when(this.requestMethodStartEvent.getUriInfo()).thenReturn(this.uriInfo);
+        when(this.responseFiltersStartEvent.getUriInfo()).thenReturn(this.uriInfo);
+        when(this.requestMethodExceptionEvent.getUriInfo()).thenReturn(this.uriInfo);
+
+        this.prepareAppEvent("methodWithDefaultAnnotation");
+    }
+
+    @Test
+    public void opensAndClosesASession() throws Exception {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://www.codacy.com/app/mitsuotakaki/CredentialStorageService-dw-hibernate/file/2455877775/issues/source?bid=3237716&fileBranchId=3250434#l98)